data/p3/prepare_p3.py

```diff
@@ -0,0 +1,439 @@
+    'wiki_hop_original_choose_best_object_affirmative_3',
+    'wiki_hop_original_choose_best_object_interrogative_1',
+    'wiki_hop_original_choose_best_object_interrogative_2',
+    'wiki_hop_original_explain_relation',
+    'wiki_hop_original_generate_object',
+    'wiki_hop_original_generate_subject',
+    'wiki_hop_original_generate_subject_and_object',
+    'wiki_qa_Decide_good_answer',
+    'wiki_qa_Direct_Answer_to_Question',
+    'wiki_qa_Generate_Question_from_Topic',
+    'wiki_qa_Is_This_True_',
+    'wiki_qa_Jeopardy_style',
+    'wiki_qa_Topic_Prediction_Answer_Only',
+    'wiki_qa_Topic_Prediction_Question_Only',
+    'wiki_qa_Topic_Prediction_Question_and_Answer_Pair',
+    'wiki_qa_automatic_system',
+    'wiki_qa_exercise',
+    'wiki_qa_found_on_google',
+    'wiqa_does_the_supposed_perturbation_have_an_effect',
+    'wiqa_effect_with_label_answer',
+    'wiqa_effect_with_string_answer',
+    'wiqa_what_is_the_final_step_of_the_following_process',
+    'wiqa_what_is_the_missing_first_step',
+    'wiqa_what_might_be_the_first_step_of_the_process',
+    'wiqa_what_might_be_the_last_step_of_the_process',
+    'wiqa_which_of_the_following_is_the_supposed_perturbation',
+    'xsum_DOC_boils_down_to_simple_idea_that',
+    'xsum_DOC_given_above_write_one_sentence',
+    'xsum_DOC_how_would_you_rephrase_few_words',
+    'xsum_DOC_tldr',
+    'xsum_DOC_write_summary_of_above',
+    'xsum_article_DOC_summary',
+    'xsum_college_roommate_asked_DOC_so_I_recap',
+    'xsum_read_below_DOC_write_abstract',
+    'xsum_summarize_DOC',
+    'xsum_summarize_this_DOC_summary',
+    'yelp_review_full_based_on_that',
+    'yelp_review_full_format_rating',
+    'yelp_review_full_format_score',
+    'yelp_review_full_format_star',
+    'yelp_review_full_on_a_scale',
+    'yelp_review_full_so_i_would',
+    'yelp_review_full_this_place'
+]
+
+# Optonally download all first
+# for task_name in TZERO_TASK_LIST:
+#     ds = load_dataset("bigscience/P3", task_name)
+
+def write_to_jsonl_hub(task_name, split):
+    # Could also use ds.to_json()
+    ds = load_dataset("bigscience/P3", task_name)
+    if split in ds:
+        with jsonlines.open(f'p3_{task_name}_{split}.jsonl', mode='w') as writer:
+            for example in ds[split].select(range(len(ds[split]))):
+                writer.write({
+                    "inputs": example["inputs_pretokenized"], 
+                    "targets": example["targets_pretokenized"]
+                })
+
+def write_to_jsonl_disk(task_name, split):
+    ds = load_from_disk(f"{os.environ['six_ALL_CCFRSCRATCH']}/datasets/p3/{task_name}")
+    if split in ds:
+        with jsonlines.open(f'p3_{task_name}_{split}.jsonl', mode='w') as writer:
+            for example in ds[split].select(range(len(ds[split]))):
+                writer.write({
+                    "inputs": example["inputs_pretokenized"], 
+                    "targets": example["targets_pretokenized"]
+                })
```

Comment:
Not sure why you have a second function. It's supposed to be the same with a different dataset loader? If so you can factorize that code.